--- mediaelement-and-player.orig.js	2015-04-28 19:38:26.000000000 -0500
+++ mediaelement-and-player.js	2015-04-28 19:30:22.000000000 -0500
@@ -12,6 +12,9 @@
  * License: MIT
  *
  */
+
+define(['jquery'], function(jQuery) {
+
 // Namespace
 var mejs = mejs || {};
 
@@ -1193,7 +1196,7 @@
       errorContainer.innerHTML = options.customError;
     } else {
       errorContainer.innerHTML = (poster !== '') ?
-        '<a href="' + playback.url + '"><img src="' + poster + '" width="100%" height="100%" /></a>' :
+        '<a href="' + playback.url + '"><img src="' + poster + '" width="100%" height="100%" alt="" /></a>' :
         '<a href="' + playback.url + '"><span>' + mejs.i18n.t('Download File') + '</span></a>';
     }
 
@@ -2931,7 +2934,7 @@
 				posterImg = posterDiv.find('img');
 
 			if (posterImg.length === 0) {
-				posterImg = $('<img width="100%" height="100%" />').appendTo(posterDiv);
+				posterImg = $('<img width="100%" height="100%" alt="" />').appendTo(posterDiv);
 			}
 
 			posterImg.attr('src', url);
@@ -3016,38 +3019,26 @@
 
 
 			// show/hide loading
-			media.addEventListener('loadeddata',function() {
-				// for some reason Chrome is firing this event
-				//if (mejs.MediaFeatures.isChrome && media.getAttribute && media.getAttribute('preload') === 'none')
-				//	return;
-
+			media.addEventListener('loadstart',function() {
 				loading.show();
 				controls.find('.mejs-time-buffering').show();
-                // Firing the 'canplay' event after a timeout which isn't getting fired on some Android 4.1 devices (https://github.com/johndyer/mediaelement/issues/1305)
-                if (mejs.MediaFeatures.isAndroid) {
-                    media.canplayTimeout = window.setTimeout(
-                        function() {
-                            if (document.createEvent) {
-                                var evt = document.createEvent('HTMLEvents');
-                                evt.initEvent('canplay', true, true);
-                                return media.dispatchEvent(evt);
-                            }
-                        }, 300
-                    );
-                }
+			}, false);
+			media.addEventListener('loadeddata',function() {
+				loading.hide();
+				controls.find('.mejs-time-buffering').hide();
 			}, false);
 			media.addEventListener('canplay',function() {
 				loading.hide();
 				controls.find('.mejs-time-buffering').hide();
-                clearTimeout(media.canplayTimeout); // Clear timeout inside 'loadeddata' to prevent 'canplay' to fire twice
 			}, false);
 
 			// error handling
-			media.addEventListener('error',function() {
+			media.addEventListener('error',function(e) {
+				t.handleError(e);
+				bigPlay.hide();
 				loading.hide();
-				controls.find('.mejs-time-buffering').hide();
 				error.show();
-				error.find('mejs-overlay-error').html("Error loading this resource");
+				error.find('.mejs-overlay-error').html("Error loading this resource");
 			}, false);
 
 			media.addEventListener('keydown', function(e) {
@@ -5486,4 +5477,6 @@
 		}
 	});
 
-})(mejs.$);
\ No newline at end of file
+})(mejs.$);
+
+});
--- css/mediaelementplayer.orig.css	2015-04-28 19:54:46.000000000 -0500
+++ css/mediaelementplayer.css	2015-04-28 19:53:39.000000000 -0500
@@ -732,6 +732,12 @@
 	padding: 15px;
 	display: block;
 }
+
+.mejs-overlay-error {
+	color: #fff;
+	text-align: center;
+	background: rgba(50, 50, 50, 0.7);
+}
 /* End: Error */
 
 
